src/neo4j_genai/retrievers/text2cypher.py

```diff
@@ -115,8 +124,9 @@ def _get_search_results(
             logger.debug("Text2CypherRetriever Cypher query: %s", t2c_query)
             records, _, _ = self.driver.execute_query(query_=t2c_query)
         except CypherSyntaxError as e:
-            logger.error("Cypher query generation failed: %s", e.message)
-            raise RuntimeError(f"Cypher query generation failed: {e.message}")
+            raise Text2CypherGenerationError(
+                f"Cypher query generation failed: {e.message}"
```

Comment:
We don't really know if the generation failed, or if it generated something that's invalid, right? Ideally, we should validate the generated query before executing it (for syntax error), but that's probably another topic. For now, I'd just rely on the python driver errors actually in this case.

Reply:
I was thinking whether `Text2CypherGenerationError` here is generic enough. Perhaps the naming is off, maybe something like `Text2CypherRetrievalError` is better?

Reply:
Another scenario could be that it generates a valid query but includes some additional junk (```, etc.) which causes the query to fail. I agree, without knowing why retrieval fails it might be best to stick to something more generic like `Text2CypherRetrievalError` as opposed to `Text2CypherGenerationError` might bias the user too much when they're trying to determine why their retriever isn't working

Reply:
Changed to `Text2CypherRetrievalError` in https://github.com/neo4j/neo4j-genai-python/pull/55/commits/53c2d3c6f32dc41425ab3adb15620eb1934a3ee6